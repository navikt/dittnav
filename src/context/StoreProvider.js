import React, { createContext, useReducer } from 'react';
import { node } from 'prop-types';
import storeReducer, { initialStoreState } from '../reducers/storeReducer';

export const StoreContext = createContext(null);

const StoreProvider = ({ children }) => {
  const [state, dispatch] = useReducer(storeReducer, initialStoreState());

  const addBeskjeder = (nyeBeskjeder) => dispatch({
    type: 'ADD_BESKJEDER',
    payload: nyeBeskjeder,
  });

  const addInaktiveBeskjeder = (nyeBeskjeder) => dispatch({
    type: 'ADD_INAKTIVE_BESKJEDER',
    payload: nyeBeskjeder,
  });

  const removeBeskjed = (beskjed) => dispatch({
    type: 'REMOVE_BESKJED',
    payload: beskjed,
  });

  const addInaktivBeskjed = (beskjed) => dispatch({
    type: 'ADD_INAKTIV_BESKJED',
    payload: beskjed,
  });

  const toggleInnloggingsModal = () => dispatch({
    type: 'TOGGLE_INNLOGGINGSMODAL',
  });

  const setError = () => dispatch({
    type: 'SET_ERROR',
  });

  return (
    <StoreContext.Provider value={{
      state,
      addBeskjeder,
      addInaktivBeskjed,
      addInaktiveBeskjeder,
      removeBeskjed,
      toggleInnloggingsModal,
      setError,
    }}
    >
      {children}
    </StoreContext.Provider>
  );
};

StoreProvider.propTypes = {
  children: node.isRequired,
};

export default StoreProvider;
