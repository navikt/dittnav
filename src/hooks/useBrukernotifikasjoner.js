import { useQuery } from 'react-query';
import { fetchBeskjeder, fetchOppgaver, fetchInnbokser } from '../Api';
import useStore from './useStore';

const useBeskjeder = () => {
  const { state, addBeskjeder, setError } = useStore();

  useQuery('beskjeder', fetchBeskjeder, {
    onSuccess: addBeskjeder,
    onError: setError,
  });

  return state.beskjeder;
};

const useOppgaver = () => (
  useQuery('oppgaver', fetchOppgaver, { onError: useStore().setError })
);

const useInnbokser = () => (
  useQuery('innbokser', fetchInnbokser, { onError: useStore().setError })
);

const useBrukernotifikasjoner = () => (
  [useBeskjeder(), useOppgaver(), useInnbokser()]
);

export default useBrukernotifikasjoner;
