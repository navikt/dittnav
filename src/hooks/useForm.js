import { useState } from 'react';

const useForm = (initialState, sendSubmit) => {
  const [state, setState] = useState(initialState);

  const handleChange = (event) => {
    event.persist();
    setState(st => ({
      ...st,
      [event.target.name]: event.target.value,
    }));
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    sendSubmit(state);
  };

  return [state, handleChange, handleSubmit];
};

export default useForm;
