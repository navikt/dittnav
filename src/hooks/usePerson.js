import { useQuery } from 'react-query';
import { fetchOppfolging, fetchMeldekort, fetchMeldinger, fetchPersonIdent, fetchPersonNavn } from '../Api';
import useStore from './useStore';

export const useOppfolging = () => (
  [useQuery('oppfolging', fetchOppfolging, { onError: useStore().setError })]
);

export const useMeldekort = () => (
  [useQuery('meldekort', fetchMeldekort, { onError: useStore().setError })]
);

export const useMeldinger = () => (
  [useQuery('meldinger', fetchMeldinger, { onError: useStore().setError })]
);

export const useIdent = (enabled) => (
  [useQuery('ident', fetchPersonIdent, { enabled, onError: useStore().setError })]
);

export const useNavn = () => (
  [useQuery('navn', fetchPersonNavn, { onError: useStore().setError })]
);

const usePerson = () => (
  [useOppfolging(), useMeldekort(), useMeldinger(), useNavn()]
);

export default usePerson;
