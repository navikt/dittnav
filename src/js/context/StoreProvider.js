import React, { createContext, useReducer } from 'react';
import { node } from 'prop-types';
import storeReducer, { initialStoreState } from '../reducers/storeReducer';

export const StoreContext = createContext(null);

const StoreProvider = ({ children }) => {
  const [state, dispatch] = useReducer(storeReducer, initialStoreState());

  const addBeskjeder = (nyeBeskjeder) => dispatch({
    type: 'ADD_BESKJEDER',
    payload: nyeBeskjeder,
  });

  const addInaktiveBeskjeder = (nyeBeskjeder) => dispatch({
    type: 'ADD_INAKTIVE_BESKJEDER',
    payload: nyeBeskjeder,
  });

  const removeBeskjed = (beskjed) => dispatch({
    type: 'REMOVE_BESKJED',
    payload: beskjed,
  });

  const addInaktivBeskjed = (beskjed) => dispatch({
    type: 'ADD_INAKTIV_BESKJED',
    payload: beskjed,
  });

  const toggleInnloggingsModal = () => dispatch({
    type: 'TOGGLE_INNLOGGINGSMODAL',
  });

  const addOppgaver = (oppgaver) => dispatch({
    type: 'ADD_OPPGAVER',
    payload: oppgaver,
  });

  const addInnbokser = (innbokser) => dispatch({
    type: 'ADD_INNBOKSER',
    payload: innbokser,
  });

  const addInaktiveOppgaver = (inaktiveOppgaver) => dispatch({
    type: 'ADD_INAKTIVE_OPPGAVER',
    payload: inaktiveOppgaver,
  });

  const addInaktiveInnbokser = (inaktiveInnbokser) => dispatch({
    type: 'ADD_INAKTIVE_INNBOKSER',
    payload: inaktiveInnbokser,
  });

  const addInnloggingsstatus = (innloggingsstatus) => dispatch({
    type: 'ADD_INNLOGGINGSSTATUS',
    payload: innloggingsstatus,
  });

  const addOppfolging = (oppfolging) => dispatch({
    type: 'ADD_OPPFOLGING',
    payload: oppfolging,
  });

  const addMeldekort = (meldekort) => dispatch({
    type: 'ADD_MELDEKORT',
    payload: meldekort,
  });

  const addNavn = (navn) => dispatch({
    type: 'ADD_NAVN',
    payload: navn,
  });

  const addIdent = (Ident) => dispatch({
    type: 'ADD_IDENT',
    payload: Ident,
  });

  const addPaabegynteSoknader = (paabegynteSoknader) => dispatch({
    type: 'ADD_PAABEGYNTESOKNADER',
    payload: paabegynteSoknader,
  });

  const addMeldinger = (meldinger) => dispatch({
    type: 'ADD_MELDINGER',
    payload: meldinger,
  });

  const addSakstema = (sakstema) => dispatch({
    type: 'ADD_SAKSTEMA',
    payload: sakstema,
  });

  const setOppfolgingFailed = () => dispatch({
    type: 'OPPFOLGING_FAILED',
  });

  const setError = () => dispatch({
    type: 'ERROR',
  });

  return (
    <StoreContext.Provider value={{
      state,
      addBeskjeder,
      addOppgaver,
      addInnbokser,
      addInaktivBeskjed,
      addInaktiveBeskjeder,
      addInnloggingsstatus,
      addInaktiveOppgaver,
      addInaktiveInnbokser,
      addOppfolging,
      addMeldekort,
      addNavn,
      addIdent,
      addPaabegynteSoknader,
      addMeldinger,
      addSakstema,
      removeBeskjed,
      toggleInnloggingsModal,
      setOppfolgingFailed,
      setError,
    }}
    >
      {children}
    </StoreContext.Provider>
  );
};

StoreProvider.propTypes = {
  children: node.isRequired,
};

export default StoreProvider;
