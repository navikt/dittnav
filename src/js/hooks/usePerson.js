import { useQuery } from 'react-query';
import Api from '../Api';

export const useOppfolging = () => (
  [useQuery('oppfolging', Api.fetchOppfolging)]
);

export const useMeldekort = () => (
  [useQuery('meldekort', Api.fetchMeldekort)]
);

export const useMeldinger = () => (
  [useQuery('meldinger', Api.fetchMeldinger)]
);

export const useNavn = () => (
  [useQuery('navn', Api.fetchPersonNavn)]
);

export const useIdent = () => (
  [useQuery('ident', Api.fetchPersonIdent)]
);
