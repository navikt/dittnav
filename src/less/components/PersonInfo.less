.person-info {
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;

  &__ikon {
    width: 2.6rem;
    height: 2.6rem;
    margin-bottom: 1rem;
  }
  h1 {
    text-transform: capitalize;
    margin: 0;
    padding-bottom: 0.5rem;
    font-size: 1.875rem;
    font-weight: 600;
  }

}
