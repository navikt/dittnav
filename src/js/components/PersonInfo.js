import React from 'react';
import Personikon from '../../assets/person.svg';
import { useIdent, useNavn } from '../hooks/usePerson';

const PersonInfo = () => {
  const [{ data: personNavn, isError: personNavnFailed }] = useNavn();
  const [{ data: personIdent, isError: personIdentFailed }] = useIdent(personNavnFailed);

  if ((!personNavn && !personIdent) || personIdentFailed) {
    return null;
  }

  const navnOrIdent = personNavnFailed ? personIdent.content.ident : personNavn.content.navn.toLowerCase();

  return (
    <div className="person-info">
      <img className="person-info__ikon" src={Personikon} alt="" />
      <h1>{navnOrIdent}</h1>
    </div>
  );
};

export default PersonInfo;
