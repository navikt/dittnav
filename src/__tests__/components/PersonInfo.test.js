import * as React from 'react';
import sinon from 'sinon';
import PersonInfo from 'js/components/PersonInfo';
import wrapIntl from 'js/utils/intlTestHelper';
import * as usePerson from '../../js/hooks/usePerson';
import StoreProvider from '../../js/context/StoreProvider';
const ReactTestRenderer = require('react-test-renderer');

let sandbox = null;

beforeEach(() => {
  sandbox = sinon.createSandbox();
});

afterEach(() => {
  usePerson.useIdent.restore();
  usePerson.useNavn.restore();
});

test('basic green PersonInfo snaphot-test', () => {
  sandbox.stub(usePerson, 'useNavn')
    .returns([{
      data: { content: { navn: 'Hello' } },
      isLoading: false,
      isError: false,
    }]);

  sandbox.stub(usePerson, 'useIdent')
    .returns([{
      data: { content: null },
      isLoading: false,
      isError: false,
    }]);

  const component = ReactTestRenderer.create(wrapIntl(
    <StoreProvider>
      <PersonInfo />
    </StoreProvider>,
  ));

  expect(component.toJSON()).toMatchSnapshot();
});

test('basic green PersonInfo snaphot-test without person and with ident', () => {
  sandbox.stub(usePerson, 'useIdent')
    .returns([{
      data: { content: { ident: 123 } },
      isLoading: false,
      isError: false,
    }]);

  sandbox.stub(usePerson, 'useNavn')
    .returns([{
      data: { content: null },
      isLoading: false,
      isError: true,
    }]);

  const component = ReactTestRenderer.create(wrapIntl(
    <StoreProvider>
      <PersonInfo />
    </StoreProvider>,
  ));

  expect(component.toJSON()).toMatchSnapshot();
});

test('PersonInfo without person and ident', () => {
  sandbox.stub(usePerson, 'useNavn')
    .returns([{
      data: { content: null },
      isLoading: false,
      isError: true,
    }]);

  sandbox.stub(usePerson, 'useIdent')
    .returns([{
      data: { content: null },
      isLoading: false,
      isError: true,
    }]);

  const component = ReactTestRenderer.create(wrapIntl(
    <StoreProvider>
      <PersonInfo />
    </StoreProvider>,
  ));

  expect(component.toJSON()).toMatchSnapshot();
});
