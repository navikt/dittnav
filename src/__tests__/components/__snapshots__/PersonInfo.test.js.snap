// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonInfo inactive 1`] = `
<div
  className="person-info"
>
  <h1
    className="person-info"
  >
    hello
  </h1>
</div>
`;

exports[`PersonInfo not registered 1`] = `
<div
  className="person-info"
>
  <h1
    className="person-info"
  >
    hello
  </h1>
</div>
`;

exports[`PersonInfo without personInfo 1`] = `null`;

exports[`basic green PersonInfo snaphot-test 1`] = `
<div
  className="person-info"
>
  <h1
    className="person-info"
  >
    hello
  </h1>
</div>
`;

exports[`basic green PersonInfo snaphot-test without fgkode/yrke 1`] = `
<div
  className="person-info"
>
  <h1
    className="person-info"
  >
    hello
  </h1>
</div>
`;
